packages/front-end/components/PowerCalculation/stats.ts

```diff
@@ -340,3 +356,294 @@ export function powerMetricWeeks(
+  nPerVariation: number,
+  relative: boolean
+): number {
+  const upperCutpoint = getCutpoint(
+    metric,
+    alpha,
+    nPerVariation,
+    relative,
+    true
+  );
+  const lowerCutpoint = getCutpoint(
+    metric,
+    alpha,
+    nPerVariation,
+    relative,
+    false
+  );
+  const powerPos = 1.0 - normal.cdf(upperCutpoint, 0, 1);
+  const powerNeg = normal.cdf(lowerCutpoint, 0, 1);
+  return powerPos + powerNeg;
+}
+
+/**
+ * Performs mde calc
+ * @param metric an object of class MetricParams that has info about prior mean and sd, metric mean and sd, and effect size.
+ * @param alpha false positive rate (default: 0.05).
+ * @param nPerVariation sample size per variation.
+ * @param relative boolean indicator if relative inference is desired.
+ * @param stepSize step size used in initial grid search.
+ * @returns object of class MDEResults, containing type and either mde or description.
+ */
+function sweepGridFine(
+  metric: MetricParams,
+  alpha: number,
+  power: number,
+  nPerVariation: number,
+  relative: boolean,
+  stepSize: number
+): MDEResults {
+  const effectSize = metric.effectSize;
+  const stepSizeFiner = stepSize / 100;
+  const lowerBound = effectSize - stepSize;
+  for (
+    let effectSizeFiner = lowerBound;
+    effectSizeFiner < effectSize;
+    effectSizeFiner += stepSizeFiner
+  ) {
+    metric.effectSize = effectSizeFiner;
+    const p = powerEstBayesian(metric, alpha, nPerVariation, relative);
+    if (p >= power) {
+      const mdeResults: MDEResults = {
+        type: "success",
+        mde: effectSizeFiner,
+      };
+      return mdeResults;
+    }
+  }
+  const mdeResults: MDEResults = {
+    type: "error",
+    description: "MDE achieving power = 0.8 does not exist in this range. ",
+  };
+  return mdeResults;
+}
+
+/**
+ * Performs mde calc
+ * @param metric an object of class MetricParams that has info about prior mean and sd, metric mean and sd, and effect size.
+ * @param alpha false positive rate (default: 0.05).
+ * @param nPerVariation sample size per variation.
+ * @param relative boolean indicator if relative inference is desired.
+ * @returns object of class MDEResults, containing type and either mde or description.
+ */
+export function findMdeBayesian(
+  metric: MetricParams,
+  alpha: number,
+  power: number,
+  nPerVariation: number,
+  relative: boolean
+): MDEResults {
+  /*fixed effect size, so prior variance of data generating process is 0*/
+  const dummyMetric = { ...metric };
+  /*first compute lower and upper bounds within stepSize*/
+  let effectSize = 0;
+  dummyMetric.effectSize = effectSize;
+  let currentPower = powerEstBayesian(
+    dummyMetric,
+    alpha,
+    nPerVariation,
+    relative
+  );
+  /*case where 0 effect size results in at least 80% power*/
+  if (currentPower >= power) {
+    const mdeResults: MDEResults = {
+      type: "success",
+      mde: effectSize,
+    };
+    return mdeResults;
+  }
+  /*maximum effect size we check is 500%*/
```

Comment:
Our FE should handle this case. If the MDE we return is 0, or the MDE we return is > 500, we just need to make sure we have tooltips or something in the FE to very lightly explain why this is.

Reply:
we can ask the customer to increase the sample size, as we currently do when mde doesn't exist.